tests/integrations/docker/test_DockerAppContext__run.py

```diff
@@ -37,12 +39,14 @@ def test_simple_call(mock_docker_app_context, tmp_path, sub_stream_kw, capsys):
     )
 
 
-def test_interactive(mock_docker_app_context, tmp_path, sub_kw, capsys):
+@pytest.mark.usefixtures("mock_docker")
+@pytest.mark.usefixtures("mock_docker_app_context")
+def test_interactive(mock_tools, my_app, tmp_path, sub_kw, capsys):
     """Docker can be invoked in interactive mode."""
-    mock_docker_app_context.run(["hello", "world"], interactive=True)
+    mock_tools[my_app].app_context.run(["hello", "world"], interactive=True)
 
     # Interactive means the call to run is direct, rather than going through Popen
-    mock_docker_app_context.tools.subprocess._subprocess.run.assert_called_once_with(
+    mock_tools.subprocess._subprocess.run.assert_called_once_with(
```

Comment:
Still have quite a bit of work to do....but was curious on your thoughts on this use of fixtures?

Several times in these tests, I'll add a fixture to the parameter list just because I need its code to run....not because I actually need a symbol for it; well, `pytest.mark.usefixture()` to the rescue.

Furthermore, though, the way I've implemented it for the `Docker` tests, it basically allows us to add tools to `mock_tools` quite nicely, IMO. I think this approach could be useful for removing blocks of code that are repeated in many related tests that effectively make a tool available or do some needed set up.

(just beware...you can't use `pytest.mark.usefixture` if the function is already also decorated with `pytest.fixture`...)

Reply:
Hrm... I think I'm a "meh". I've kind of internalised the "a fixture is something that runs, but doesn't require you to use it" thinking, so I'm not sure what we gain out of having a separate `use_fixtures` decorator. I don't fundamentally object to the pattern - I just don't see what we get from using it - especially since it moves fixture naming to be a string, rather than a symbol.